tests/Doctrine/ODM/MongoDB/Tests/Proxy/Factory/StaticProxyFactoryTest.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ODM\MongoDB\Tests\Proxy\Factory;
+
+use Doctrine\ODM\MongoDB\DocumentManager;
+use Doctrine\ODM\MongoDB\LockException;
+use Doctrine\ODM\MongoDB\Proxy\Factory\StaticProxyFactory;
+use Doctrine\ODM\MongoDB\Tests\BaseTestCase;
+use Documents\PaymentProvider;
+use MongoDB\Client;
+use MongoDB\Collection;
+use MongoDB\Database;
+use PHPUnit\Framework\MockObject\MockObject;
+
+class StaticProxyFactoryTest extends BaseTestCase
+{
+    /** @var Client|MockObject */
+    private Client $client;
+
+    private StaticProxyFactory $proxyFactory;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->dm           = $this->createMockedDocumentManager();
+        $this->proxyFactory = new StaticProxyFactory($this->dm);
+    }
+
+    public function testProxyInitializeWithException(): void
+    {
+        $collection = $this->createMock(Collection::class);
+        $database   = $this->createMock(Database::class);
+
+        $this->client->expects($this->once())
+            ->method('selectDatabase')
+            ->willReturn($database);
+
+        $database->expects($this->once())
+            ->method('selectCollection')
+            ->willReturn($collection);
+
+        $collection->expects($this->once())
+            ->method('findOne')
+            ->willThrowException(LockException::lockFailed(null));
+
+        $uow = $this->dm->getUnitOfWork();
+
+        $metadata = $this->dm->getClassMetadata(PaymentProvider::class);
+
+        $proxy = $this->proxyFactory->getProxy($metadata, '123');
+        $uow->registerManaged($proxy, '123', []);
+
+        try {
+            $proxy->initializeProxy();
```

Comment:
```suggestion
            $proxy->initializeProxy();
            self::fail('An exception should have been thrown');
```

Let's save us from any future refactorings that may change expectations on how the exception is thrown.